config.sample.yaml

```diff
@@ -35,9 +35,10 @@ ldap:
       # explicitly. Most of the time it will work either way, but if
       # this is not set any non-UTF8 values will fail to sync.
       is_binary: true
-    status: "shadowInactive"
-    enable_value: 512
-    disable_value: 514
+    # Status flag for the user. This is read as a big-endian integer. Normally userAccountControl in AD.
+    status: "shadowFlag"
+    # Vector of biflag values that marks the user as disabled. Tested on status. (ACCOUNTDISABLE=2 in AD)
+    disable_bitflags: [2]
```

Comment:
If we want to match against a bitmask, shouldn't we be parsing a single hex value here instead of a vec?

Reply:
I thought it is easier to configure the invidual flags rather than putting in one bitmask

Reply:
Is this supposed to be the vector of bit indexes? So `[2]` means `0b01` (or `0b001`?)? 
From the code where `&` happens it looks like all the values in this vec are `&`-ed, then I'm not really getting why this is vector

Reply:
Hope now it is more clear with a second flag as an example LOCKOUT=16. Each flag is tested individual against the status.

Reply:
so, 16 would be 0b10000 mask? or 0b10000000000000000? It seems the first case, but maybe it could be commented with some examples?

Reply:
or maybe rename to `disable_bitmasks` because these seems to be masks, and "bitflags" make me think that the value in the list might be not mask, but a flag, and flag could be represented as the position of the bit (1 << 0 = 0b1, 1 << 1 = 0b10, 1 << 2 = 0b100), and that position could start from the 1 (1 << (1-1) = 0b1, 1 << (2-1) = 0b10, 1 << (3-1) = 0b100), so very ambiguous

Reply:
I think it would add more confusion if I would name it bitmask. It should be clear after the given examples and a look of the other flags what is expected https://learn.microsoft.com/en-us/troubleshoot/windows-server/active-directory/useraccountcontrol-manipulate-account-properties.

Reply:
Each value in the list **can** be a bitmask, bitmask **can** include **several** flags. If `disable_bitflags` would not be a list, then there wouldn't be any confusion, as "bitflag**s**" ~ "bitmask". So when I see `s` ending and the type of the value is a list, I strip the `s` and the root of the word should be the type of the elements of the list. Following this logic, the list of `disable_bitflags` should be `disable_bitflag`, stripping the adjective -- just `bitflag`. However bitflag is just 0 or 1, basically boolean, but what is inside `disable_bitflags` list is not that, it is not 0 or 1, it is a bitmask, i.e. many bitflags.

Reply:
I think I'd prefer if this was just a simple bitmask expressed as a hexadecimal integer. Interacting with account status requires understanding bitmasks anyway, having a different format than what people use for other queries is more confusing IMO.

Reply:
> having a different format than what people use for other queries is more confusing

I agree on this, but I don't see where the format is different. I'm merely suggesting to rename a variable to reflect more the underlying values. 

Anyway I don't want this small issue to stall the development process, but if you have time, please gather other opinions from @famedly/workflows

Reply:
Settled on disable_bitmasks